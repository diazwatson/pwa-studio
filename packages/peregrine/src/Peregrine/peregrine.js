import React from 'react';
import { arrayOf, oneOf } from 'prop-types';
import { ToastContextProvider } from '../Toasts';
import { WindowSizeContextProvider } from '../hooks/useWindowSize';

const contextProviders = {
    TOASTS: ToastContextProvider,
    WINDOW_SIZE: WindowSizeContextProvider
};

const Peregrine = ({ children, providers }) => {
    return providers.reduceRight((memo, providerKey) => {
        const ContextProvider = contextProviders[providerKey];
        return <ContextProvider>{memo}</ContextProvider>;
    }, children);
};

Peregrine.propTypes = {
    providers: arrayOf(oneOf(Object.keys(contextProviders)))
};

Peregrine.defaultProps = {
    providers: ['WINDOW_SIZE', 'TOASTS']
};

export default Peregrine;
