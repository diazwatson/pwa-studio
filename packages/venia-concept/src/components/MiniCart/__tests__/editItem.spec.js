import React, { useEffect } from 'react';
import ShallowRenderer from 'react-test-renderer/Shallow';
import { useQuery } from '@magento/peregrine';

import EditItem from '../editItem';

jest.mock('react', () => {
    const React = jest.requireActual('react');
    const spy = jest.spyOn(React, 'useEffect');

    return Object.assign(React, { useEffect: spy });
});
jest.mock('@magento/peregrine');

const renderer = new ShallowRenderer();

test('renders null when item not supplied', () => {
    const tree = renderer.render(<EditItem />);

    expect(tree).toMatchSnapshot();
});

test('renders cart options when item has no options', () => {
    const props = {
        item: {
            options: []
        }
    };

    const tree = renderer.render(<EditItem {...props} />);

    expect(tree).toMatchSnapshot();
});

test('renders a loading indicator while running query', () => {
    const props = {
        item: {
            options: ['a', 'b', 'c']
        }
    };

    const tree = renderer.render(<EditItem {...props} />);

    expect(tree).toMatchSnapshot();
});

test('renders cart options when item has options', () => {
    const queryApi = {
        runQuery: jest.fn(),
        setLoading: jest.fn()
    };
    const queryResult = {
        data: {
            products: {
                items: []
            }
        }
    };
    useQuery.mockReturnValueOnce([queryResult, queryApi]);
    useEffect.mockReturnValueOnce(undefined);

    const props = {
        item: {
            options: ['a', 'b', 'c']
        }
    };

    const tree = renderer.render(<EditItem {...props} />);

    expect(tree).toMatchSnapshot();
});
