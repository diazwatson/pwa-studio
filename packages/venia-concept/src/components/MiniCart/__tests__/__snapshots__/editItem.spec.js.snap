// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a loading indicator while running query 1`] = `
<Classify(LoadingIndicator)>
  <span>
    Fetching Data...
  </span>
</Classify(LoadingIndicator)>
`;

exports[`renders cart options when item has no options 1`] = `
<CartOptions
  cartItem={
    Object {
      "options": Array [],
    }
  }
  configItem={Object {}}
/>
`;

exports[`renders cart options when item has options 1`] = `
<CartOptions
  cartItem={
    Object {
      "options": Array [
        "a",
        "b",
        "c",
      ],
    }
  }
/>
`;

exports[`renders null when item not supplied 1`] = `null`;
